### ISSUE TITLE
Provide an alternative format for error/warning messages; eg: file:line:col instead of file(line, col)

### ISSUE THREAD

**Issue by ziotom78** [nim-lang/Nim#690]
Currently the Nimrod compiler produces error messages using a format string similar to the one used by Free Pascal, i.e., with the line and column numbers enclosed in parentheses:

```
  test.nim(5, 0) Error: invalid indentation
```

However, this kind of syntax is not recognized by Emacs. When you run the compiler using Emacs' `compile` function, you cannot highlight the errors and move from one to another using `next-error`, because such functions expect something in the form

```
  test.nim:5:0: Error: invalid indentation
```

(see http://www.gnu.org/prep/standards/html_node/Errors.html#Errors).

The Free Pascal compiler `fpc` solves this by accepting the (scarcely documented) command-line flag `-vr` ("Rhide/GCC compatibility mode"):

```
  $ fpc test.pas
  ...
  test.pas(4,6) Error: Identifier not found "X"
  ...
  $ fpc -vr test.pas
  ...
  test.pas:4: error 6: Error: Identifier not found "X"
  ...
```

which is not perfect but at least allows Emacs to find the right line in the source code.

Since I am an avid Emacs user, would somebody be interested in having this feature implemented in Nimrod? I can fork the repository and try to implement it by myself, but first I would know if it is sensible to add such a patch, and if there is anything I must be aware of.

**Comment by ghost** [comment]
Still would be nice.

**Comment by timotheecour** [comment]
I refactored all (hopefully) file/line/col formatting used by compiler in a single proc in lib/std/private/miscdollars.nim so this should now be much easier to change (see https://github.com/nim-lang/Nim/pull/8795 for my earlier attempt)

the way I'd like this to work is that passing a compiler option would affect both compiler messages as well as standard runtime messages (assertion backtraces), without having to recompile nim. Many ways to achieve this that are low in complexity.

**Comment by lhupfeldt** [comment]
Yes, please let us have an option for file:line:col. This is probably the most universal error format.
